tests/integration/test_trading.py

```diff
@@ -294,3 +299,795 @@ def test_estimate_position(vega_service_with_market: VegaServiceNull):
+            market_id=market_id,
+        )
+ 
+    open_volume_A = PARTY_A_position.open_volume
+    risk_factors = vega.get_risk_factors(market_id=market_id)
+    risk_factor = (
+        risk_factors.long if open_volume_A > 0 else risk_factors.short
+    )
+
+    liquidation_price_open_vol_only_best_case = max((collateral_available - open_volume_A * market_data.mark_price)/(open_volume_A * 0 + open_volume_A**2 * 0 + open_volume_A * risk_factor - open_volume_A),0)
+    liquidation_price_open_vol_only_worst_case = max((collateral_available - open_volume_A * market_data.mark_price)/(open_volume_A * linear_slippage_factor + open_volume_A**2 * quadratic_slippage_factor + open_volume_A * risk_factor - open_volume_A),0)
+    
+    # #check the calculation of estimate_liquidation_price_open_vol_only.best_case.open_volume_only
+    assert round(estimate_liquidation_price_open_vol_only.best_case.open_volume_only,12)== round(liquidation_price_open_vol_only_best_case,12)
+    assert round(estimate_liquidation_price_open_vol_only.worst_case.open_volume_only,12)== round(liquidation_price_open_vol_only_worst_case,12)
+
+    #amend order in order to set the new mark price 
+    vega.amend_order(
+        trading_key=PARTY_C.name,
+        market_id=market_id,
+        order_id=order_id_C,
+        price=985,
+        volume_delta=20,
+    )
+    #change the mark price to closeout PARTY_A
+    vega.submit_order(
+        trading_key=PARTY_C.name,
+        market_id=market_id,
+        order_type="TYPE_LIMIT",
+        time_in_force="TIME_IN_FORCE_GTC",
+        side="SIDE_BUY",
+        order_ref="new_buy_price_to_trade",
+        price=985,
+        volume=initial_volume,
+    )
+    vega.submit_order(
+        trading_key=PARTY_B.name,
+        market_id=market_id,
+        order_type="TYPE_LIMIT",
+        time_in_force="TIME_IN_FORCE_GTC",
+        side="SIDE_SELL",
+        order_ref="new_sell_price_to_trade",
+        price=985,
+        volume=initial_volume,
+    )
+    vega.wait_fn(1)
+    vega.wait_for_total_catchup()
+    #AC 0012-NP-LIPE-001:Check actual liquidation price is within the estimated price 
+    print(f"market after amending at new price {vega.get_latest_market_data(market_id=market_id)}")
+    print(f"account_PARTY_A is closeout since sum of account = {vega.party_account(key_name=PARTY_A.name, asset_id=asset_id, market_id=market_id).general+vega.party_account(key_name=PARTY_A.name, asset_id=asset_id, market_id=market_id).margin}")
+    assert vega.party_account(key_name=PARTY_A.name, asset_id=asset_id, market_id=market_id).general+vega.party_account(key_name=PARTY_A.name, asset_id=asset_id, market_id=market_id).margin ==0
+    assert estimate_liquidation_price_open_vol_only.best_case.open_volume_only >= vega.get_latest_market_data(market_id=market_id).mark_price
+    assert estimate_liquidation_price_open_vol_only.worst_case.open_volume_only<= vega.get_latest_market_data(market_id=market_id).mark_price
```

Comment:
There are a couple of places here where we can reduce the number of data-node queries we use by making the call once and then using the response in each of the assertions.